Dockerfile

```diff
@@ -20,84 +20,81 @@ ARG COMMIT_SHA
 ARG TARGETPLATFORM
 
 COPY docker/install_build_dependencies.sh /tmp/
-RUN chmod +x /tmp/install_build_dependencies.sh && /tmp/install_build_dependencies.sh
-
 COPY docker/go_wrapper.sh /go/bin/go
-RUN chmod +x /go/bin/go
+COPY api /app
+WORKDIR /app
+
 ENV GOPATH="/go"
 ENV PATH="${GOPATH}/bin:${PATH}"
-
 ENV CGO_ENABLED 1
 
-RUN go env
-
-RUN mkdir -p /app
-WORKDIR /app
-
 # Download dependencies
-COPY api/go.mod api/go.sum /app/
-RUN go mod download
-
-# Patch go-face
-RUN sed -i 's/-march=native//g' ${GOPATH}/pkg/mod/github.com/!kagami/go-face*/face.go
-
-# Build dependencies that use CGO
-RUN go install \
-  github.com/mattn/go-sqlite3 \
-  github.com/Kagami/go-face
-
-# Copy and build api source
-COPY api /app
-RUN go build -v -o photoview .
+RUN chmod +x /tmp/install_build_dependencies.sh \
+  && chmod +x /go/bin/go \
+  && /tmp/install_build_dependencies.sh \
+  && go env \
+  && go mod download \
+  # Patch go-face
+  && sed -i 's/-march=native//g' ${GOPATH}/pkg/mod/github.com/!kagami/go-face*/face.go \
+  # Build dependencies that use CGO
+  && go install \
+    github.com/mattn/go-sqlite3 \
+    github.com/Kagami/go-face \
+  # Build api source
+  && go build -v -o photoview .
 
 ### Copy api and ui to production environment ###
-FROM debian:bookworm
+FROM debian:bookworm-slim
 ARG TARGETPLATFORM
-WORKDIR /app
 
-COPY api/data /app/data
-
-RUN apt update \
+# Create a user to run Photoview server
+RUN useradd -r -U -m photoview \
   # Required dependencies
-  && apt install -y curl gpg libdlib19.1 ffmpeg exiftool libheif1
-
-# Install Darktable if building for a supported architecture
-RUN if [ "${TARGETPLATFORM}" = "linux/amd64" ] || [ "${TARGETPLATFORM}" = "linux/arm64" ]; then \
-  apt install -y darktable; fi
-
-# Remove build dependencies and cleanup
-RUN apt purge -y gpg \
-  && apt autoremove -y \
-  && apt clean \
+  && apt-get update \
+  && apt-get install -y curl gnupg gpg libdlib19.1 ffmpeg exiftool libheif1 sqlite3 \
+  # Install Darktable if building for a supported architecture
+  && if [ "${TARGETPLATFORM}" = "linux/amd64" ] || [ "${TARGETPLATFORM}" = "linux/arm64" ]; then \
+    echo 'deb https://download.opensuse.org/repositories/graphics:/darktable/Debian_12/ /' \
+      | tee /etc/apt/sources.list.d/graphics:darktable.list; \
+    curl -fsSL https://download.opensuse.org/repositories/graphics:/darktable/Debian_12/Release.key \
+      | gpg --dearmor | tee /etc/apt/trusted.gpg.d/graphics_darktable.gpg > /dev/null; \
+    apt-get update; \
+    apt-get install -y darktable; \
+  fi \
+  # Remove build dependencies and cleanup
+  && apt-get purge -y gnupg gpg \
+  && apt-get autoremove -y \
+  && apt-get clean \
   && rm -rf /var/lib/apt/lists/*
 
-COPY --from=ui /app/dist /ui
+WORKDIR /home/photoview
+COPY api/data /app/data
+COPY --from=ui /app/dist /app/ui
 COPY --from=api /app/photoview /app/photoview
 
 ENV PHOTOVIEW_LISTEN_IP 127.0.0.1
-ENV PHOTOVIEW_LISTEN_PORT 80
+ENV PHOTOVIEW_LISTEN_PORT 8080
```

Comment:
it is required by the "run as non-root user" improvement. We discussed this here: https://github.com/photoview/photoview/pull/863#discussion_r1576311663

Reply:
Are you sure that this is the case for Docker environments as well?
I just found this post: https://forums.docker.com/t/non-root-user-able-to-bind-to-port-80-why/140524

Maybe we should make an issue keeping track of the changes that we want to make for version 3.0?

Reply:
We discussed this as part of the @Omar007 port change proposal. While I can confirm that the recent version of Docker allows using 80 port by a non-root user inside a container, it is not clear, starting from which version of Docker this change is introduced.
I can assume that some Linux distros or devices with proprietary Linux distros might still provide an older Docker version, where this is not yet implemented.
So, making sure that Photoview works according to Linux best practices on all supported platforms, looks more beneficial than keeping port 80 with the hope that it will work correctly everywhere.
I don't see a negative impact of switching the port number. Please share your thoughts if there is something negative, related to this change.

Reply:
I see @kkovaletp covered the reasoning already but @viktorstrate is also technically correct in that it can be 'breaking'.  
If a user had been running as root or granted the required capabilities (or had been on a Docker version greater than ...?) and has never explicitly set `PHOTOVIEW_LISTEN_PORT` themselves at any point and where to just swap out the image, it would be 'breaking' in the sense that their port-forwards/mapping would stop working if they don't explicitly set `PHOTOVIEW_LISTEN_PORT=80` when they update.  

Since this is a defaults change but also one that is backward compatible (it does not require these users to change assigned permissions or capabilites; they could start dropping these now but they don't need to), I wouldn't say this would require a major version bump, postponing to a major release or anything like that, but I would at the very least make sure this is mentioned on w/e the tag ends up being.

Reply:
Thanks @Omar007 for the explanation!
If we review the case, where a user updates an image, but doesn't update the compose and .env files (which doesn't make much sense from fixes and improvements delivery, but can happen in reality) nothing is going to break, as the compose file from the current `master` contains the `PHOTOVIEW_LISTEN_PORT=80`: https://github.com/photoview/photoview/blob/4133694bc23d53db3fa484d04283e75a57ef00db/docker-compose.example.yml#L27
Of course, if this line was removed or commented out, it wouldn't work, but that is not the recommended config.

BTW, while answering this comment, I noticed the hardcoded port in the `EXPOSE` and `HEALTHCHECK`, so I replaced it with the variable, then tested that it works fine with the `8080` and `80` values (setting `80` in the compose file for the same image)